### PR TITLE
add rsh port flag

### PR DESCRIPTION
**Pull request by plazmakeks**
in case a private key file is supplied a port can be defined which shall be used to establish a connection via ssh. used for testing with mock ssh servers.

### CODE REVIEW

**Comment by gchamon** [comment]
I believe `-p` takes precedence over `.ssh/config`. This being the case, a configuration that already specifies a port would be overwritten by the default configuration.

I believe port and strict host checking, just like private_key, should be optional and only produce effect on the outputted command if specified.

Another thing is that port and strict host checking are dependent on `private_key` being specified, which could be counter intuitive. I don't have a better option to model this option for the current API, and I believe this PR is important. I will give this some more thought and get back to you.

**Comment by plazmakeks** [comment]
Hey @gchamon, i made some refactoring due to your comments

**Comment by gchamon** [comment]
hey there, sorry for the extremely long wait
I will merge to a secondary branch, resolve the conflicts, do a little refactoring that I don't feel like imposing to contributors, and I'll release it!
Thanks a lot

File: sysrsync/runner.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def run(cwd=os.getcwd(), strict=True, verbose=False, **kwargs):
     if verbose is True:
         print(f'[sysrsync runner] running command on "{cwd}":')
         print(rsync_string)
-    process = subprocess.run(rsync_string, cwd=cwd, shell=True)
+    process = subprocess.run(rsync_string, cwd=cwd, shell=True, stdout=subprocess.PIPE)
```
Comment (gchamon) [inline sysrsync/runner.py:15]: @plazmakeks one question, why should we need to use PIPE here?

Comment (gchamon) [reply sysrsync/runner.py:15]: since I am removing shell in the next patch, I am removing stdout too

**Pull request closed without merging**